Kerberos.NET/Crypto/Pal/Windows/BCryptAlgorithmProviderCache.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Concurrent;
+
+namespace Kerberos.NET.Crypto.Pal.Windows
+{
+    internal static class BCryptAlgorithmProviderCache
```

Comment:
According the [docs](https://docs.microsoft.com/en-us/windows/win32/api/bcrypt/nf-bcrypt-bcryptopenalgorithmprovider#remarks) `BCryptOpenAlgorithmProvider` is very costly, so the handles get cached.

Ideally `BCryptCloseAlgorithmProvider` should be called with the handle when it's no longer in use.
I didn't find any information whether it's harmful if not called or not. In CoreFx is not called in the cache either, so I assume this shouldn't be a problem.